conda_build/os_utils/liefldd.py

```diff
@@ -238,12 +238,10 @@ def get_uniqueness_key(file):
          and  # noqa
          (binary.type == lief.ELF.ELF_CLASS.CLASS32 or binary.type == lief.ELF.ELF_CLASS.CLASS64)):
         dynamic_entries = binary.dynamic_entries
-        for e in dynamic_entries:
-            result = [e.name for e in dynamic_entries if e.tag == lief.ELF.DYNAMIC_TAGS.SONAME]
-            if result:
-                return result[0]
-            return binary.name
-        return binary.soname
```

Comment:
Go compiled binaries do not have dynamic_entries, so the for loop never executed, which caused line 246 to be triggered instead of 245.

If you look closely at lines 242-245 you will notice that it does not depend on the outside iterated variable `e` anyways and instead was relying on the list-comprehension in 242.